### DISCUSSION TITLE
Skill for Alexa

### DISCUSSION THREAD

**Discussion by Alexandrump** [nukeop/nuclear#1180] [Ideas]
Just a happy thought! But, maybe it could be take into consideration for the future.

I was thinking on the possibility of adding Nuclear music player as a service, for example, running on a raspberry pi, and then configure it for running like a server, via url and port. Then, like [Plex does](https://www.techhive.com/article/583189/how-to-play-music-you-own-on-an-amazon-echo.html), having a [skill for Alexa](https://developer.amazon.com/es-ES/alexa/alexa-skills-kit), that links ours local network discovering the service and then giving us the possibility to talk with Alexa to search and play songs.

I've never worked with Amazon Skills, and I don't know if this could be done. The main reason is about AWS TOS, but probably someone knows more about this.

Thank you for reading guys!

**Comment by nukeop** [comment]
There is already a server that lets you control Nuclear to some extent, but it doesn't stream music to external consumers. Do people listen to music on Alexa? If you want to just control it, you can use the API we already provide and hook the skill to these endpoints.

**Reply by Alexandrump** [reply]
Yes, that's interesting, the possibility to control the player via voice in Alexa, but as you say, I go further. What I'm proposing is to stream music to external consumers, and I don't know if this could be done with the curren API?

Ansewering to your question. Yes, I often play music on Alexa with other music apps. And probably a poll, give us more responses about if it could be a good feature.

In other hand, to run Nuclear without running a graphical interface, I was thinking if Nuclear could be used in CLI (not GUI) mode? Because it will give us the ability to run in "server mode", using less resources, and then all the development for Raspberry Pi (or similars) could have sense.

I will explore the AWS Alexa Developer Kit, and I would like to hear if there is interest about this functionality.
What do you think guys?

**Comment by Alexandrump** [comment]
I've just published a poll to observe if this have your interest. https://nuclear.featureupvote.com/suggestions/276891/skill-for-alexa

:)
